go/controller/internal/reconciler/reconciler.go

```diff
@@ -333,59 +333,69 @@ func (a *kagentReconciler) ReconcileKagentMCPServer(ctx context.Context, req ctr
 }
 
 func (a *kagentReconciler) ReconcileKagentRemoteMCPServer(ctx context.Context, req ctrl.Request) error {
-	// reconcile the agent team itself
-	toolServer := &v1alpha2.RemoteMCPServer{}
-	if err := a.kube.Get(ctx, req.NamespacedName, toolServer); err != nil {
-		// if the tool server is not found, we can ignore it
+	serverRef := req.NamespacedName.String()
+	l := reconcileLog.WithValues("remoteMCPServer", serverRef)
+
+	server := &v1alpha2.RemoteMCPServer{}
+	if err := a.kube.Get(ctx, req.NamespacedName, server); err != nil {
+		// if the remote MCP server is not found, we can ignore it
 		if k8s_errors.IsNotFound(err) {
 			// Delete from DB if the remote mcp server is deleted
 			dbServer := &database.ToolServer{
-				Name:      utils.GetObjectRef(toolServer),
-				GroupKind: schema.GroupKind{Group: "kagent.dev", Kind: "RemoteMCPServer"}.String(),
+				Name:      serverRef,
+				GroupKind: server.GroupVersionKind().GroupKind().String(),
 			}
+
 			if err := a.dbClient.DeleteToolServer(dbServer.Name, dbServer.GroupKind); err != nil {
-				reconcileLog.Error(err, "failed to delete tool server for remote mcp server", "remoteMCPServer", utils.GetObjectRef(toolServer))
```

Comment:
`toolServer` will be empty at this point since we failed to get it - hence the broader refactoring to set `serverRef` at the start of the function and re-use that everywhere instead of calling `utils.GetObjectRef` multiple times.